src/Symfony/Component/Console/Input/InputOption.php

```diff
@@ -69,7 +69,7 @@ public function __construct(string $name, $shortcut = null, int $mode = null, st
             throw new InvalidArgumentException('An option name cannot be empty.');
         }
 
-        if (empty($shortcut)) {
+        if ('' === $shortcut || [] === $shortcut) {
```

Comment:
$shortcut === false is not handled

Reply:
can you please send a PR on branch 5.4 to fix this?

Reply:
The documented acceptable types for `$shortcut` are `string`, `array`, and `null`, and the previously-existing test cases only covered those types, so in that sense passing `false` here is an incorrect use of the API in the first place. 

But passing `false` happened to accidentally work before the change, so I guess it is a breaking change (to undocumented behavior).

A fix that restores the handling of `false` should probably also update the types in the docstring and add a test case, so that the next person who has reason to modify this code knows that behavior should be preserved.